### PR TITLE
[T16] Create API route to create a RESCHEDULING_REQUEST

### PR DESCRIPTION
**Pull request by absirgue**
Hey dear beautiful ppl,

Finished my ticket for creating rescheduling request and would like ur feedback and to see what can be improved :)

A bit of context bcs some of it gets tricky:
- The request is given a new_date parameter in the format MM/DD/YYYY 
- This new date parameter in fact influences 3 of the columns of the ReschedulingRequest record that will be created
                       the NewWeek (week number of the week the new class will happen in)
                       the NewDay (the day of the week this new class will happen in)
                       the NewDate (the new date as such)
NewWeek is calculated using  functions in helper_functions.js that were merged in T14 and that allow to calculate the number of days that passed since the start of a week we know the number of. We divide that number by 7 and it will give us the number of weeks that elapsed.
NewDay is calculated pretty simply bcs JS allows to do it easily.

### CODE REVIEW

**Comment by absirgue** [comment]
Btw, I know logins are not hidden, I'll hide them whenever u both agree and I merge bcs otherwise it gets shitty

**Review by Ali-Alkhars — CHANGES_REQUESTED** [review]
Very clever implementation Monsieur! There are some things that need to be addressed though:

1. Add a guard for null "new_date".
2. Shouldn't accept dates in the past or today.
3. Should check the format of the given date, if not in "MM/DD/YYYY" an error with a unique status should be returned so the bot could inform the users. (The bot can't force users to use a specific text entry format).

File: helper_functions.js [hunk 1]
```diff
@@ -72,4 +25,180 @@ module.exports = {
...
+    } else {
+      if (toTestMonth < todayMonth) {
+        return false;
+      } else if (toTestMonth > toTestMonth) {
+        return true;
+      } else {
+        if (toTestDay > todayDay) {
+          return true;
+        } else {
+          return false;
```
Comment (Ali-Alkhars) [inline helper_functions.js:148]: You could make it
```
if (toTestYear < todayYear) {
      return false;
    } else if (toTestYear > todayYear) {
      return true;
    } else {
      if (toTestMonth < todayMonth) {
        return false;
      } else if (toTestMonth > toTestMonth) {
        return true;
      } else {
        return toTestDay > todayDay;
        }
      }
    }
```

File: helper_functions.js [hunk 2]
```diff
@@ -72,4 +25,180 @@ module.exports = {
...
+    if (!date.includes("/")) {
+      return false;
+    }
+    const splittedDate = date.split("/");
+    if (!splittedDate.length === 3) {
+      return false;
+    } else {
+      const numericalDay = parseInt(splittedDate[1]);
+      const numericalMonth = parseInt(splittedDate[0]);
+      if (
```
Comment (Ali-Alkhars) [inline helper_functions.js:104]: you could make it
```
if (!splittedDate.length === 3) {
      return false;
    } else {
      const numericalDay = parseInt(splittedDate[1]);
      const numericalMonth = parseInt(splittedDate[0]);
      return
        splittedDate[0].length == 2 &&
        splittedDate[1].length == 2 &&
        splittedDate[2].length == 4 &&
        numericalDay < 32 &&
        numericalMonth < 13;
    }
```

**Review by Ali-Alkhars — APPROVED** [review]
Perfect Anton! I left two code syntax suggestions 🦊🦊

File: helper_functions.js (hunk 2 above)
Comment (absirgue) [reply helper_functions.js]: Yep both are great ideas, thanks! I updated it

**Review by TunaAtasayar — APPROVED** [review]
Well Done!

**Pull request merged**